pkg/asset/installconfig/vsphere/validation.go

```diff
@@ -77,7 +78,12 @@ func folderExists(finder Finder, ic *types.InstallConfig, fldPath *field.Path) f
 	return nil
 }
 
-func validateNetwork(finder Finder, p *vsphere.Platform, fldPath *field.Path) field.ErrorList {
+func validateNetwork(client *vim25.Client, finder *find.Finder, p *vsphere.Platform, fldPath *field.Path) field.ErrorList {
+	// It's not possible to validate a network if datacenter or cluster are empty strings
+	if p.Datacenter == "" || p.Cluster == "" {
```

Comment:
I'll take a look a this. This was added due to a failing test, but the test can be adjusted instead of adding this check.

Reply:
Looking at this again, without this check in place, it would result in unfriendly errors like this when the Cluster isn't specified in install config:
```
FATAL failed to fetch Terraform Variables: failed to fetch dependency of "Terraform Variables": failed to generate asset "Platform Provisioning Check": [platform.vsphere.cluster: Required value: must specify the cluster, platform.vsphere.network: Invalid value: "TestDC": datacenter './TestDC' not found] 
```

_With_ these changes, the error shown in case of missing cluster will remain the same and much more user friendly:
```
FATAL failed to fetch Terraform Variables: failed to fetch dependency of "Terraform Variables": failed to generate asset "Platform Provisioning Check": platform.vsphere.cluster: Required value: must specify the cluster 
```